Mathlib/LinearAlgebra/DirectSum/Finsupp.lean

```diff
@@ -88,4 +92,109 @@ theorem finsuppTensorFinsupp'_single_tmul_single (a : ι) (b : κ) (r₁ r₂ :
   aesop (add norm [Finsupp.single_apply])
 #align finsupp_tensor_finsupp'_single_tmul_single finsuppTensorFinsupp'_single_tmul_single
 
+end TensorProduct
+
+/-!
+The case of `PiTensorProduct`.
+-/
+
+section PiTensorProduct
+
+open PiTensorProduct BigOperators
+
+attribute [local ext] TensorProduct.ext
+
+variable (R : Type*) [CommSemiring R]
+
+variable {ι : Type*}
+
+variable [Fintype ι]
+
+variable [DecidableEq ι]
+
+variable (κ : ι → Type*) [(i : ι) → DecidableEq (κ i)]
+
+variable (M : ι → Type*)
+
+variable [∀ i, AddCommMonoid (M i)]
+
+variable [∀ i, Module R (M i)]
+
+variable [(i : ι) → (x : M i) → Decidable (x ≠ 0)]
```

Comment:
Note that the style has now changed to omit these extra blank lines; it was only ever a result of `mathport` anyway.
```suggestion
variable (R : Type*) [CommSemiring R]
variable {ι : Type*}
variable [Fintype ι]
variable [DecidableEq ι]
variable (κ : ι → Type*) [(i : ι) → DecidableEq (κ i)]
variable (M : ι → Type*)
variable [∀ i, AddCommMonoid (M i)]
variable [∀ i, Module R (M i)]
variable [(i : ι) → (x : M i) → Decidable (x ≠ 0)]
```